conftest.py

```diff
@@ -0,0 +1,120 @@
+from pathlib import Path
+import subprocess
+from typing import Union
+
+from py._path.local import LocalPath
+import pytest
+from pytest import Session
+
+
+EXAMPLES_DIR = Path(__file__).parent / "examples"
+
+
+class ExampleFailedException(Exception):
+    """Raised when an example fails."""
+
+    def __init__(self, message: str, exit_status: int):
+        super().__init__(message)
+        self.exit_status = exit_status
+
+
+class ExampleRunner:
+    """Run the docker-compose of a given example."""
+
+    def __init__(self, compose_file: str):
+        self.compose_file = compose_file
+
+    def compose(self, *command: str) -> int:
+        """Runs docker-compose using subprocess with the given command.
+
+        Returns exit status and output.
+        """
+        try:
+            subprocess.run(
+                ["docker-compose", "-f", self.compose_file, *command],
```

Comment:
on my m1 system, I run `docker compose` not `docker-compose` on the command line.

Reply:
Ah, right. Fun lol. I'll try to figure out if there's a straightforward way to account for that.

Reply:
https://docs.docker.com/compose/migrate/#docker-compose-vs-docker-compose

Reply:
Maybe a silly comment along these lines, @dbluhm wasn't this mentioned during the ACA-Pug meeting today? Might be able to be reconciled in the future (very possible I'm off on this)

Reply:
I'm thinking it might be simplest to just alias `docker-compose` for `docker compose` locally if you want to run these directly. Else, we can always just let the action run it and not worry about local environments.

Reply:
Not sure I heard mention of docker-compose vs docker compose on the ACA-PUG call today :thinking:

Reply:
Ah, no wait you're right, I was confusing it with the amd vs. arm thingy/separate thing, my bad